flexmeasures/data/models/time_series.py

```diff
@@ -51,6 +56,92 @@ def search_beliefs(
             source=source,
         )
 
+    def chart(
+        self,
+        events_not_before: Optional[datetime_type] = None,
+        events_before: Optional[datetime_type] = None,
+        belief_start: Optional[datetime_type] = None,
+        belief_end: Optional[datetime_type] = None,
+        source: Optional[Union[int, List[int], str, List[str]]] = None,
+        chart_type: str = "bar_chart",
+        data_only: bool = False,
+        chart_only: bool = True,
+        as_html: bool = False,
+        dataset_name: Optional[str] = None,
+        **kwargs,
+    ) -> str:
+        """
+
+        :param data_only: return just the data (in case you have the chart specs already)
+        :param as_html: return the chart with data as a standalone html
+        """
+
+        # Set up chart specification
+        if dataset_name is None:
+            dataset_name = "sensor_" + str(self.id)
+        self.sensor_type = (
+            self.name
+        )  # todo remove this placeholder when sensor types are modelled
+        chart = belief_charts_mapping[chart_type](
+            title=capitalize(self.name),
+            quantity=capitalize(self.sensor_type),
+            unit=self.unit,
+            dataset_name=dataset_name,
+            **kwargs,
+        )
+        if chart_only:
+            return json.dumps(chart)
```

Comment:
So as far as I understand I would implement your first idea as follows:

- `with_data_as=None` returns the chart specs only, as a json string
- `with_data_as="part-of-chart"` returns the chart specs incl. data, as a json string
- `with_data_as="standalone-json"` returns the data only, as a json string
- `with_data_as="html-page"` returns the chart specs incl. data, as a html string

But then I don't understand the need for your second idea, because "standalone-json" would already do that.

Reply:
If we also go for the second idea, we indeed don't need `with_data_as="standalone-json"`. Instead, that would become its own endpoint. One that we already have, in a way (` GET /api/v2_0/getMeterData`).

Here we could start to discuss what happens to the likes of getMeterData ― will they stay, and we also add a REST-like alternative (`GET /sensor/{id}/data`)? But I realise we don't have to decided that right now. But creating a corner case where people can do exactly what they can do with another endpoint seems to me like smelly design.

Reply:
The standalone data returned by this new /sensors/<id>/chart endpoint is not exactly the data returned by the old getMeterData endpoint. It's very different actually: quite raw beliefsdataframe contents specifically meant to support FM chart creation for any sensor, as opposed to carefully curated power values in a USEF-coated message meant to support cross-platform communication of meter readings.